sys/fs/cd9660/cd9660_vnops.c

```diff
@@ -140,8 +144,25 @@ cd9660_access(struct vop_access_args *ap)
 		}
 	}
 
-	return (vaccess(vp->v_type, ip->inode.iso_mode, ip->inode.iso_uid,
-	    ip->inode.iso_gid, ap->a_accmode, ap->a_cred));
+	file_mode = ip->inode.iso_mode;
+	switch (vp->v_type) {
```

Comment:
If you wanted, you could perhaps simplify the `switch` down to an if.

I looked at msdosfs a bit more and what it does is to set default values of its equivalent of `im_dmask` and `im_fmask` so that it can use them unconditionally without needing flags.  You could choose to do that here as well by initializing `im_dmask` and `im_fmask` to `ACCESSPERMS` in `cd9660_mount` before you check for the options.  You could them remove the `ISOFSMNT_DMASK` and `ISOFSMNT_FMASK` flags and replace this switch with something like `file_mode &= (vp->v_type == VDIR ? ip->i_mnt->im_dmask : ip->i_mnt->im_fmask;` both here and in the `VOP_GETATTR`.

What you have now is also correct, but making that change might be more concise.

Reply:
Brilliant!  I followed your suggestions and successfully tested a RR & non-RR image with & without the introduced options and it seems to work.